common/kvstore/tablestore/table_store_builder.go

```diff
@@ -0,0 +1,374 @@
+package tablestore
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/common/kvstore"
+	"github.com/Layr-Labs/eigenda/common/kvstore/leveldb"
+	"github.com/Layr-Labs/eigenda/common/kvstore/mapstore"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"math"
+	"sort"
+)
+
+// The table ID reserved for the metadata table.
+const metadataTableID uint32 = math.MaxUint32
+
+// The table ID reserved for the namespace table. Keys in the namespace table are table IDs (uint32)
+// and values are table names (string).
+const namespaceTableID uint32 = math.MaxUint32 - 1
+
+// This key is used to store the schema version in the metadata table.
+const metadataSchemaVersionKey = "schema_version"
+
+// The metadataDeletionKey holds the value of the table name that is being deleted if one is currently being deleted.
+// This is required for atomic deletion of tables.
+const metadataDeletionKey = "deletion"
+
+// The current schema version of the metadata table.
+const currentSchemaVersion uint64 = 0
+
+// StoreType describes the underlying store implementation.
+type StoreType int
+
+const (
+	// LevelDB is a LevelDB-backed store.
+	LevelDB StoreType = iota
+	// MapStore is an in-memory store. This store does not preserve data across restarts.
+	MapStore
+)
+
+// Create creates a new TableStore instance of the given type. The store will be created at the given path.
+// This method will set up a table for each table name provided, and will drop all tables not in the list.
+// Dropping a table is irreversible and will delete all data in the table, so be very careful not to call
+// this method with table names omitted by mistake.
+func (t StoreType) Create(logger logging.Logger, path string, tables ...string) (kvstore.TableStore, error) {
```

Comment:
It looks a more frequent operation will be `Load`, where it's given the path, and will load the existing base Store as well as existing tables.

Reply:
I've renamed this to `Start()`, since that name is accurate for both a new setup and for when we load existing data.

Reply:
Potential UX issues to check: 1) the users mostly may just want to load existing tables (adding/dropping tables are rare operations), and 2) the user may not know the existing tables, unless they can build this TableStore, which then requires knowing the existing tables (a loop here)

Reply:
I added a method `Load()` which can be used to load the store without making any modifications to the schema.